test/Microsoft.ML.Tests/Transformers/NormalizerTests.cs

```diff
@@ -240,6 +244,87 @@ public void SimpleConstructorsAndExtensions()
             Done();
         }
 
+        [Fact]
+        public void SupervisedBin()
+        {
+            string dataPath = GetDataPath(TestDatasets.iris.trainFilename);
+
+            var loader = new TextLoader(Env, new TextLoader.Arguments
+            {
+                Column = new[] {
+                    new TextLoader.Column("Label", DataKind.R4, 0),
+                    new TextLoader.Column("float4", DataKind.R4, new[]{new TextLoader.Range(1, 4) }),
+                }
+            });
+
+            var data = loader.Read(dataPath);
+
+            //var est1 = new NormalizingEstimator(Env, "float4");
+            //var trans1 = est1.Fit(data);
+            //var data1 = trans1.Transform(data);
+
+            var est2 = new NormalizingEstimator(Env, NormalizingEstimator.NormalizerMode.MinMax, ("float4", "float4"));
+            var trans2 = est2.Fit(data);
+            var data2 = trans2.Transform(data);
+
+            //var est3 = new NormalizingEstimator(Env, new NormalizingEstimator.MinMaxColumn("float4"));
+            //var trans3 = est3.Fit(data);
+            //var data3 = trans3.Transform(data);
+
+            var est4 = new NormalizingEstimator(Env,
+                new NormalizingEstimator.SupervisedBinningColumn("float4", "float4_1"),
+                new NormalizingEstimator.SupervisedBinningColumn("float4", "float4_2"));
+            var trans4 = est4.Fit(data);
+            var data4 = trans4.Transform(data);
+            // 'transformedData' is a 'promise' of data. Let's actually read it.
+            var someRows = data4.AsEnumerable<OutputScore>(Env, reuseRowObject: false);
+
+            // Extract the 'AllFeatures' column.
+            // This will give the entire dataset: make sure to only take several row
+            // in case the dataset is huge. The is similar to the static API, except
+            // you have to specify the column name and type.
+            var featureColumns = data4.GetColumn<float[]>(Env, "float4_1");
+
+            var cursorable = data4.AsCursorable<OutputScore>(Env);
+            using (var cursor = cursorable.GetCursor())
+            {
+                // We are now in charge of creating the row object.
+                var myRow = new OutputScore();
+                while (cursor.MoveNext())
+                {
+                    // Populate the values of the row object.
+                    cursor.FillValues(myRow);
+                    var name = myRow.float4_1;
+                }
+            }
+
+            data4.Schema.TryGetColumnIndex("float4_1", out int output);
+            using (var cursor = data4.GetRowCursor(col => col == output))
+            {
+                var buffer = default(VBuffer<float>);
+                var getter = cursor.GetGetter<VBuffer<float>>(output);
+                var numRows = 0;
+                while (cursor.MoveNext())
+                {
+                    getter(ref buffer);
+                    numRows += 1;
```

Comment:
>numRows [](start = 20, length = 7)

do you need to validate the data?

Reply:
For now I have appended to the existing test `SimpleConstructorsAndExtensions`  to verify schema and values. 

Once this PR goes in, I intent to enable the skipped tests in TLC and possibly port over some TLC tests as well. But that would be a separate PR.

---
In reply to: [241653196](https://github.com/dotnet/machinelearning/pull/1876#discussion_r241653196) [](ancestors = 241653196)